gst/nnstreamer/nnstreamer_plugin_api_converter.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * NNStreamer API for Tensor_Decoder Sub-Plugins
+ * Copyright (C) 2019 MyungJoo Ham <myungjoo.ham@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file  nnstreamer_plugin_api_converter.h
+ * @date  09 Dec 2019
+ * @brief Mandatory APIs for NNStreamer Converter sub-plugins (Need Gst Devel)
+ * @see https://github.com/nnsuite/nnstreamer
+ * @author  MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#ifndef __NNS_PLUGIN_API_CONVERTER_H__
+#define __NNS_PLUGIN_API_CONVERTER_H__
+
+#include "tensor_typedef.h"
+#include <gst/gst.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/***********************************************
+* External Converters                          *
+************************************************/
+
+/**
+ * @brief Converter's subplugin implementation.
+ */
+struct _NNStreamerExternalConverter {
+  const char *media_type_name;
+
+  /** 1. chain func, data handling. */
+  GstBuffer * (*convert) (GstTensorConverter * self,
+      const GstBuffer * buf, gsize * frame_size, guint * frames_in);
+      /**< Convert the given input stream to tensor/tensors stream.
+       * @param[in/out] self A pointer designating "this".
+       * @param[in] buf The input stream buffer
+       * @param[out] frame_size The size of each frame (output buffer)
+       * @param[out] frames_in The number of frames in the given input buffer.
+       * @retval Return inbuf if the data is to be kept untouched.
+       * @retval Retrun a new GstBuf if the data is to be modified.
+       */
+
+  /** 2. parse_caps (type conv, input(media) to output(tensor)) */
+  gboolean (*get_caps) (GstTensorConverter * self, const GstStructure * st,
+      GstTensorConfig * config);
+      /**< Set the tensor config structure from the given stream frame
+       * @param[in/out] self A pointer designating "this".
+       * @param[in] st The input (original/media data) stream's metadata
+       * @param[out] config The output (tensor/tensors) emtadata
```

Comment:
If you are considering multi-tensors (later work), lets pass `GstTensorsConfig`.

Reply:
TODO: testcases with extern-converter, need improve with properties.